src/compile/axis/parse.ts

```diff
@@ -242,7 +242,11 @@ function getProperty<K extends keyof AxisComponentProps>(property: K, specifiedA
       return numberFormat(fieldDef, specifiedAxis.format, model.config);
     case 'grid': {
       const scaleType = model.getScaleComponent(channel).get('type');
-      return getSpecifiedOrDefaultValue(specifiedAxis.grid, properties.grid(scaleType, fieldDef));
+      if (model.specifiedScales[channel].binned) {
```

Comment:
Hmm if binned is specified via `"bin"`, should `binned` really still be a part of the specified scale?  (Esp. that we still have access to the fieldDef here.)

Reply:
Ah just notice that you're implementing `scale.binned = true`, not `bin: 'binned'` that we finally agreed on.